js/views/file_input_view.js

```diff
@@ -10,22 +10,30 @@ var Whisper = Whisper || {};
         },
 
         events: {
-            'change': 'previewImages'
+            'change': 'previewImages',
+            'click': 'deleteFiles'
         },
 
         addThumb: function(e) {
-          this.$el.append(
-            $('<img>').attr( "src", e.target.result ).addClass('preview')
-          );
+            var divToAdd = $('<div>').addClass('imageAttachment');
+            this.$el.append(divToAdd);
+            divToAdd.append($('<img>').attr('src', e.target.result).addClass('preview'));
+            divToAdd.append($('<div>').addClass('close').html('x'));
         },
 
         previewImages: function() {
-            this.$el.find('img').remove();
+            this.clearForm();
             var files = this.$input.prop('files');
             for (var i = 0; i < files.length; i++) {
                 var FR = new FileReader();
-                FR.onload = this.addThumb.bind(this);
-                FR.readAsDataURL(files[i]);
+                if (((files[i].size/1024)/1024).toFixed(4) >= 10) {
```

Comment:
The "official" recommendation is a 420K limit for images, mostly as a courtesy to mobile clients with limited storage and bandwidth...